### PR TITLE
Add py-atlannot version 0.1.2

### PR DESCRIPTION
**Pull request by Stannislav**
* The new version (`0.1.2`) is hosted on GitHub / PyPI (open-sourced recently)
* The old version (`0.1.1`) is still pointing to GitLab because there won't be a release for it on PyPI.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-atlannot/package.py [hunk 1]
```diff
@@ -10,29 +10,36 @@
...
+    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools-scm", type="build")
 
     # Installation requirements
-    depends_on('py-antspyx@0.2.4', type=('run'))
-    depends_on('py-atldld@0.2.2', type=('run'))
-    # depends_on('py-dvc', type=('run'))
-    depends_on('py-matplotlib', type=('run'))
-    depends_on('py-numpy', type=('run'))
+    depends_on("py-antspyx@0.2.4", type=("run"))
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-atlannot/package.py]: All dependencies here and below should actually be `"build", "run"`

Comment (Stannislav) [reply var/spack/repos/builtin/packages/py-atlannot/package.py]: Is that something new? Weren't all normal python dependencies usually just listed with `type=("run")`?

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-atlannot/package.py]: Didn't pay much attention to it before (and used both normally), but Gianluca pointed out at some point: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#python-dependencies

> These packages are required for building and installation. You can add them with type=('build', 'run').

Kind of a safeguard against `pip` downloading anything required during installation with Spack.

Comment (Stannislav) [reply var/spack/repos/builtin/packages/py-atlannot/package.py]: Fair enough, thanks for the reference.

Changed in 1c493704d7

**Review by matz-e — APPROVED** [review]

**Pull request merged**